node/packages/parser/src/ethereum/yearn.ts

```diff
@@ -0,0 +1,65 @@
+import { Tx } from '@shapeshiftoss/blockbook'
+import { GenericParser, TxSpecific, YearnTx } from '../types'
+import shapeShiftRouter from './abi/shapeShiftRouter'
+import yearnVault from './abi/yearnVault'
+import { ethers } from 'ethers'
+import { SHAPE_SHIFT_ROUTER_CONTRACT } from './constants'
+import { getSigHash, getYearnTokenVaultAddresses } from './utils'
+
+export class Parser implements GenericParser {
+  yearnTokenVaultAddresses: Array<string> | undefined
+
+  async parse(tx: Tx): Promise<TxSpecific<YearnTx> | undefined> {
+    const data = tx.ethereumSpecific?.data
+    if (!data) return
+
+    const abiInterface = this.getAbiInterface(data)
```

Comment:
Might as well return early if no abiInterface and then we can reduce the following optional chaining.

Reply:
Addressed in https://github.com/shapeshift/unchained/pull/267/commits/5fecfd3a2687cfdb37e4b0d571c38d4bcca7b628

Reply:
Addressed in https://github.com/shapeshift/unchained/pull/267/commits/5fecfd3a2687cfdb37e4b0d571c38d4bcca7b628